### ISSUE TITLE
Footnotes are not inserted correctly

### ISSUE THREAD

**Issue by openmedi** [SublimeText-Markdown/MarkdownEditing#317]
I am writing a paper that is around 30 pages and devide my time writing this thing between an editorial/iPad and a MacBook/SublimeText(3) Setup. I'm using OS X 10.11.2 and some pandoc and academic writing related plugins in addition to Markdown Editing.

In ST3 with MarkdownEditing (MDE) I have the problem that footnote insertation doesn't honor the format the footnotes are in, meaning, that when I'm inserting a new footnotes via the ```cmd+shift+6``` shortcut a footnote indeed gets inserted, but it is just appended to the end of the file while also adding a space in front of the line, without regard for the other footnotes. I assume detecting this kind of stuff is dificult and might not be something that MDE wants to implement, I want to suggest though, that at least the space in front of the footnotes that are inserted in that way is removed.

Another thing is, that MDE seems to be unable to find the appropriate footnotes, when using the above mentioned command on an allready existing footnote. This seems to be a hard to reproduce phaenomenon but it is present in my current document, although not with all footnotes. I understand that a foonotes numbers are more ore less reference strings and their order shoudln't matter, I therefore never bothered to make an effort to order those after their IDs.

So, in short:

1. there is a space added to the "automatic" footnotes, which seems wrong; also it would be nice if footnotes could get their own paragraph and it would be even niver (although not very likely to be included) if MDE could check what style of footnote formating is actually in use right now
2. using the footnote shortcut on already existing footnotes not always yields the expected results more often than not I try to jump between the footnote and its contents and instead get transported to a completely different footnote

**Comment by felixhao28** [comment]
Thank you for the feedback. Actually, all commands related to footnotes, references, links, images have been reorganized and rewritten in v2.1.5 (the upcoming version). If you need it now, you can try following manual installation guide. Remember to `git checkout v2.1.5` after cloning. When v2.1.5 is finally out, you can safely remove the folder (backup user settings!) and use Package Control.

> there is a space added to the "automatic" footnotes, which seems wrong;

I agree. And the extraneous space is removed in v2.1.5.

> it would be nice if footnotes could get their own paragraph

That is a reasonable request. I will think about that. #306 is talking about the exact same thing.

>  it would be even nicer if MDE could check what style of footnote formating is actually in use right now

In v2.1.5, there is a "organize reference" command that gathers and alphabetically sorts footnotes and references.

> using the footnote shortcut on already existing footnotes not always yields the expected results more often than not I try to jump between the footnote and its contents and instead get transported to a completely different footnote

The `cmd+shift+6` command is changed to insertion only in v2.1.5. To jump around, use `cmd+ctrl+shift+l` or "Jump Reference" command.
